overlord/devicestate/devicemgr.go

```diff
@@ -1543,6 +1543,55 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) hasUserExpired(user *auth.UserState) bool {
+	// If the user has no expiration date, then Expiration should not
+	// be set, and contain the default value.
+	if user.Expiration.IsZero() {
+		return false
+	}
+
+	return user.Expiration.Before(time.Now())
+}
+
+// ensureExpiredUsersRemoved is periodically called as a part of Ensure()
+// to remove expired users from the system.
+func (m *DeviceManager) ensureExpiredUsersRemoved() error {
+	st := m.state
+	st.Lock()
+	defer st.Unlock()
+
+	// XXX: Only in run mode?
```

Comment:
This comment and the one below are questions, yet describe exactly what the code is doing. If you want to draw the reviewer's attention, I would write a comment here in github.

At this point I think you can remove the two comments; if you decide to keep them anyways, I'd add a bit of more information in them, to explain why we might want to do otherwise.

Reply:
Good point, updated the comments. Thanks!